### PR TITLE
UX-524/Displayed region using 'DisplayName' instead of 'RegionName' to be more user friendly

### PR DESCRIPTION
**Pull request by knnguy**
Screen Shot 2020-09-14 at 2 16 00 PM

### CODE REVIEW

File: src/app/plugins/kubernetes/components/common/CloudProviderRegionPicklist.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { loadCloudProviderDetails } from 'k8s/components/infrastructure/cloudPro
 // We need to use `forwardRef` as a workaround of an issue with material-ui Tooltip https://github.com/gregnb/mui-datatables/issues/595
 const CloudProviderRegionPicklist = forwardRef(({ cloudProviderId, ...rest }, ref) => {
   const [details, loading] = useDataLoader(loadCloudProviderDetails, { cloudProviderId })
-  const options = pluck('RegionName', details)
+  const options = pluck('DisplayName', details)
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/common/CloudProviderRegionPicklist.js:11]: Does this work with other cloud providers? aka I believe when the type is azure we use `DisplayName` when its aws we use `RegionName`

File: src/app/plugins/kubernetes/components/common/CloudProviderRegionPicklist.js [hunk 2]
```diff
@@ -6,9 +6,10 @@ import { pluck } from 'ramda'
 import { loadCloudProviderDetails } from 'k8s/components/infrastructure/cloudProviders/actions'
 
 // We need to use `forwardRef` as a workaround of an issue with material-ui Tooltip https://github.com/gregnb/mui-datatables/issues/595
-const CloudProviderRegionPicklist = forwardRef(({ cloudProviderId, ...rest }, ref) => {
+const CloudProviderRegionPicklist = forwardRef(({ cloudProviderId, type, ...rest }, ref) => {
   const [details, loading] = useDataLoader(loadCloudProviderDetails, { cloudProviderId })
-  const options = pluck('RegionName', details)
+  const displayField = type === 'aws' ? 'RegionName' : 'DisplayName'
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/common/CloudProviderRegionPicklist.js:11]: Lets do the inverse of this. meaning the one off / edge case is Azure needs to use `DisplayName`. We should check explicitly for azure. `type === 'azure'  ? 'DisplayName' : 'RegionName'`

File: src/app/plugins/kubernetes/components/infrastructure/clusters/AddAwsClusterPage.js [hunk 3]
```diff
@@ -497,6 +497,7 @@ const AddAwsClusterPage = () => {
                             cloudProviderRegionId={wizardContext.cloudProviderRegionId}
                             onChange={(value) => setWizardContext({ azs: value })}
                             values={wizardContext.azs}
+                            type="aws"
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/AddAwsClusterPage.js:500]: treat `type` here as a required property and pass it here and also on the azure form

File: src/app/plugins/kubernetes/components/common/CloudProviderRegionPicklist.js [hunk 4]
```diff
@@ -6,9 +6,10 @@ import { pluck } from 'ramda'
 import { loadCloudProviderDetails } from 'k8s/components/infrastructure/cloudProviders/actions'
 
 // We need to use `forwardRef` as a workaround of an issue with material-ui Tooltip https://github.com/gregnb/mui-datatables/issues/595
-const CloudProviderRegionPicklist = forwardRef(({ cloudProviderId, ...rest }, ref) => {
+const CloudProviderRegionPicklist = forwardRef(({ cloudProviderId, type, ...rest }, ref) => {
```
Comment (jruddell) [inline src/app/plugins/kubernetes/components/common/CloudProviderRegionPicklist.js:9]: Please put a comment above this component with a todo `// TODO use the CloudProviders enum to strongly type this type field once moved to TS`

File: src/app/plugins/kubernetes/components/infrastructure/clusters/AddAwsClusterPage.js (hunk 3 above)
Comment (knnguy) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/AddAwsClusterPage.js:500]: `type` is already passed to the CloudProviderRegionPicklist in the azure form
Screen Shot 2020-09-16 at 2 52 30 PM

**Review by jruddell — APPROVED** [review]

**Pull request merged**